apis/eventing/v1alpha1/cloudeventsource_types.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2023 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudEventSource defines how a keda event will be sent to event sink
+// +kubebuilder:resource:path=cloudeventsources,scope=Namespaced
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Active",type="string",JSONPath=".status.conditions[?(@.type==\"Active\")].status"
+type CloudEventSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CloudEventSourceSpec   `json:"spec"`
+	Status CloudEventSourceStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// CloudEventSourceList is a list of CloudEventSource resources
+type CloudEventSourceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []CloudEventSource `json:"items"`
+}
+
+// CloudEventSourceSpec defines the spec of CloudEventSource
+type CloudEventSourceSpec struct {
+	// +optional
+	ClusterName string `json:"clusterName,omitempty"`
```

Comment:
Just to verify : This is optional and can be configured in KEDA deployment itself, right?

Reply:
Yes.

Reply:
@josefkarasek https://github.com/kedacore/keda/pull/4968#discussion_r1334944831